Parsers/ASimFileEvent/Parsers/vimFileEventAzureBlobStorage.yaml

```diff
@@ -27,35 +73,72 @@ ParserQuery: |
       "QueryBlobContents", "FileAccessed",
       "GetBlob", "FileAccessed",
       "AppendBlock", "FileModified",
-      "ClearPage",  "FileModified",
+      "ClearPage", "FileModified",
       "PutBlockFromURL", "FileModified",
-      "DeleteBlob",  "FileDeleted",
-      "DeleteContainer",  "FolderDeleted"
-      ];
+      "DeleteBlob", "FileDeleted",
+      "DeleteContainer", "FolderDeleted"
+  ];
       StorageBlobLogs
+      | where not(disabled)
+      | where (isnull(starttime) or TimeGenerated >= starttime) 
+          and (isnull(endtime) or TimeGenerated <= endtime)
       // **** relevant data filtering;
       | where OperationName in (bloboperations)
+      | where ((array_length(srcipaddr_has_any_prefix) == 0) or (has_any_ipv4_prefix(CallerIpAddress, srcipaddr_has_any_prefix))) and 
+          (array_length(actorusername_has_any) == 0) and
+          ((array_length(targetfilepath_has_any) == 0) or (Uri has_any (targetfilepath_has_any))) and 
+          (array_length(srcfilepath_has_any) == 0) and
```

Comment:
@anki-narravula 
The SAAS service products like blob, file, sharepoint, are not providing these field values. Instead of Actor, they log SrcIpAddr. EDR products mostly are logging those fields.